src/main/java/uk/gov/MintApplication.java

```diff
@@ -76,6 +78,31 @@ public void run(MintConfiguration configuration, Environment environment) throws
                                         .buildAuthFilter()
                         ))
                 );
+
+        startHeartbeating(configuration);
+    }
+
+    private void startHeartbeating(MintConfiguration configuration) {
+        if (configuration.getCloudWatchEnvironmentName().isPresent()) {
+            heartbeatThread = new Thread(new CloudWatchHeartbeater(configuration.getCloudWatchEnvironmentName().get(), configuration.getRegister()));
+            heartbeatThread.start();
+        } else {
+            heartbeatThread = null;
+        }
+
```

Comment:
This could be simplified by using java.util.concurrent and dropwizard's managed executors:

```java
        ScheduledExecutorService cloudwatch = environment.lifecycle().scheduledExecutorService("cloudwatch").threads(1).build();
        cloudwatch.scheduleAtFixedRate(() -> {}, 0, 10000, TimeUnit.MILLISECONDS);
```

You would also change the CloudWatchHeartbeater to only perform a single heartbeat.  The ScheduledExecutorService will handle repeating the task.

By using the executorservice from the dropwizard `environment`, the executorservice will be automatically shut down when the app is shut down.